forms-flow-web/src/components/Form/Item/Edit.js

```diff
@@ -693,24 +750,84 @@ const Edit = React.memo(() => {
       </div>
     );
   }
+  const unPublishActiveForm = ()=> {
+    if (processListData.status === "active") {
+      unPublish(processListData.id)
+        .then(() => {
+          setIsPublishLoading(false);
+          setIsPublished(!isPublished);
+          dispatch(push(`${redirectUrl}form`));
+        })
+        .catch((err) => {
+          setIsPublishLoading(false);
+          const error = err.response?.data || err.message;
+          dispatch(setFormFailureErrorData("form", error));
+        });
+  }};
+
+  const handleCloseActionModal = () => {
+    setSelectedAction(null); // Reset action
+  };
+
+  const renderDeleteModal = () => {
+    const hasSubmissions = processListData.id && applicationCount;
+    if (hasSubmissions) {
+      // Case when the form has associated submissions and cannot be deleted
+      return (
+        <ConfirmModal
+          show={selectedAction === DELETE}
+          title={t("You Cannot Delete This Form")}
+          message={t(
+            "But you may unpublish it if you wish to not receive any more submissions."
+          )}
+          messageSecondary={t(
+            "You may not delete a form that has submissions associated with it."
+          )}
+          primaryBtnAction={handleCloseActionModal}
+          onClose={handleCloseActionModal}
+          secondayBtnAction={unPublishActiveForm}
+          primaryBtnText={t("Keep This Form")}
+          secondaryBtnText={t("Unpublish This Form")}
+          size="md"
+        />
+      );
+    } else {
+      // Case when the form can be deleted
+      return (
+        <ConfirmModal
+          show={selectedAction === DELETE}
+          title={t("Are You Sure You Want to Delete This Form?")}
+          message={t("This action cannot be undone.")}
+          primaryBtnAction={handleCloseActionModal}
+          onClose={handleCloseActionModal}
+          secondayBtnAction={deleteModal}
+          primaryBtnText={t("No, Keep This Form")}
+          secondaryBtnText={t("Yes, Delete the Form")}
+          size="md"
```

Comment:
I have doubt ,do we have size prop for confirm .it is fixed right?. @abilpraju-aot